src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+
+
+        private const ParseFlags ParseFlagsDateMask = ParseFlags.HaveHour | ParseFlags.HaveMinute | ParseFlags.HaveSecond | ParseFlags.HaveTime | ParseFlags.TimeZoneUsed |
+                                                      ParseFlags.TimeZoneUtc | ParseFlags.CaptureOffset | ParseFlags.UtcSortPattern;
+
+        /// <summary>
+        /// Converts a memory span that contains string representation of a date to its DateOnly equivalent by using culture-specific format information and a formatting style.
+        /// </summary>
+        /// <param name="s">The memory span that contains the string to parse.</param>
+        /// <param name="provider">An object that supplies culture-specific format information about s.</param>
+        /// <param name="style">A bitwise combination of enumeration values that indicates the permitted format of s. A typical value to specify is None.</param>
+        /// <returns>An object that is equivalent to the date contained in s, as specified by provider and styles.</returns>
+        public static DateOnly Parse(ReadOnlySpan<char> s, IFormatProvider? provider, DateTimeStyles style = DateTimeStyles.None)
+        {
+            if ((style & ~DateTimeStyles.AllowWhiteSpaces) != 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidDateStyles, nameof(style));
+            }
+
+            DateTimeResult result = default;       // The buffer to store the parsing result.
+            result.Init(s);
+
+            if (!DateTimeParse.TryParse(s, DateTimeFormatInfo.GetInstance(provider), style, ref result))
+            {
+                throw new FormatException(SR.Format(SR.Format_BadDateOnly, s.ToString()));
+            }
+
+            if ((result.flags & ParseFlagsDateMask) != 0)
+            {
+                throw new FormatException(SR.Format(SR.Format_DateOnlyContainsNoneDateParts, s.ToString()));
+            }
+
+            return new DateOnly(DayNumberFromDateTime(result.parsedDate));
+        }
+
+        /// <summary>
+        /// Converts the specified span representation of a date to its DateOnly equivalent using the specified format.
+        /// The format of the string representation must match the specified format exactly or an exception is thrown.
+        /// </summary>
+        /// <param name="s">A span containing the characters that represent a date to convert.</param>
+        /// <param name="format">A span containing the characters that represent a format specifier that defines the required format of s.</param>
+        /// <returns>An object that is equivalent to the date contained in s, as specified by format.</returns>
+        public static DateOnly ParseExact(ReadOnlySpan<char> s, ReadOnlySpan<char> format) => ParseExact(s, format, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None);
+
+        private const string OFormat = "yyyy'-'MM'-'dd";
+        private const string RFormat = "ddd, dd MMM yyyy";
```

Comment:
Make sense to move to top of the file?

Reply:
I want to keep the definition closer to where we use it.